feat(redis): add faq for redis (#708)
scaleway/docs-content: faq/databases-for-redis.mdx

---
meta:
  title: Databases for Redis™
  description: Discover Scaleway Managed Databases and how to access it for Redis™.
content:
  h1: Databases for Redis™
  paragraph: Discover Scaleway Managed Databases and how to access it for Redis™.
image: database-redis.webp
---

## What is Scaleway Databases for Redis™

Databases for Redis™ is a low-latency caching solution. It allows you to easily set up a secure cache and lighten the load on your main database. Based on the in-memory data storage, Databases for Redis™ improves your application response time and helps you provide a better experience to your users. 

> The resource optimizes the speed of your requests as copies of the most frequently used data are stored in memory making them accessible in milliseconds. Learn more about Redis™ databases.

Reviewer: I feel like "The resource" is not clear enough, maybe "Caching data" or "Using Database for Redis as a cache"

## Why is it ideal for Cache usage?

> Based on the **Re**mote **Di**ctionary **S**erver technology, Scaleway Database for Redis™ stores your data in the RAM of the underlying machine rather than on a disk (SSD/HDD). In other words, for each request to read, to insert, or to update data in a database, this can be executed using data available in the fastest and closest storage of your compute resource, the memory. Traditional databases like MySQL or PostgreSQL store data on a disk which inevitably introduces IOPs and results in latency on each operation. Redis™ is known for delivering millisecond response time and high performance for millions of requests per second to empower demanding workloads. The combination of a powerful in-memory data storage such as Redis and the management of the resources set-up, securisation, scaling and maintenance makes Scaleway Database for Redis a handy solution to improve the usability of your application 

Reviewer: "s" missing in Scaleway Database**s** for Redis™

One of the most common ways to implement cache is storing frequently accessed data in Redis™ (therefore in memory), and serving your application's request from there, should data be available, instead of from your traditional database. If data is not available in memory, it can always be retrieved from the primary database. Below is a simplified schema explaining this process . 

> ## How can I access my Redis once it is provisioned?

Reviewer: TM missing on Redis

> You can connect to your Redis™  using the IP address.es and port.s available in the details page of your Database for Redis™ Instance next to the title **Endpoints**.

Reviewer: Here we can simply use RedisTM Instance instead of Database for Redis™ Instance

- There are as many `IP:port` as nodes in your cluster (one if standalone).
- If you are in cluster mode (3 nodes or more), you have to use a cluster specific connector able to connect to multiple endpoints.

ADD SCREENSHOT.

> ## What is the Private Networks feature for Database Instances?

Reviewer: RedisTM Instances instead of Database Instances?

> [Private Networks](https://www.scaleway.com/en/docs/network/vpc/quickstart/) allows you to configure your own Layer-2 network with IPv4 addresses on Database for Redis™ Instances. 

Reviewer: s missing in Database**s** for RedisTM

> You can therefore enhance the security of your system architecture by isolating it from the Internet. It also increases the security of your databases, since Instances in your Private Network can directly be connected to your Redis™ Instance, without passing through the public Internet.

Reviewer: I feel like we are saying the same thing twice

- In standalone mode, you can create new Redis™ Instances to attach to your Private Network, or attach existing Database Instances by adding Private Networks endpoints to them. 
- In standalone mode, you can attach Private Networks to your Redis™ Instances at the creation level or add a Private Networks endpoint to your existing Redis™ Instance from the console.

<Message type="note">

> Bear in mind that in cluster mode, Private Network is only available at the creation of the cluster Instance and cannot be modified afterwards. If you want to switch back to a public network or update your IP adresses, you have to create a new Redis Instance. 

Reviewer: Sometime Private Network is singular, others plural
I'd say `you will have to create a new RedisTM Instance`

</Message>

## How many Private Networks can I attach?
 
Currently, you can connect one Private Network to each Database for Redis Instance in the Scaleway Console. For Standalone Instances you can attach multiple Private Networks via the CLI or the API only.

## Can I use any IP Range for my Private Network?
 

> For now, only the [RFC1918](https://datatracker.ietf.org/doc/html/rfc1918) IP subnets are accepted as a Database for Redis Instance virtual IPv4.

Reviewer: Simply `RedisTM Instance` instead of `Database for Redis Instance`

## How is the Redis Database's failover IP managed?

> The Redis controller is aware of the cluster topology and routes requests automatically to a healthy node. In case of unavailability of one of the cluster nodes, no downtime is to be expected.

Reviewer: In cluster mode (3 nodes and above), the RedisTM controller is aware of the cluster topology and routes requests automatically to a healthy node. In case of unavailability of one of the cluster nodes, no downtime is to be expected.

You can scale your cluster horizontally up to six nodes. Below is an example of configuration for a three nodes cluster:
* Instance A contains hash slots from 0 to 5,500
* Instance B contains hash slots from 5,501 to 11,000
* Instance C contains hash slots from 11,001 to 16,383

Each of the three Instances acts as a primary node and replicates one of the others as a secondary node.

## What's the difference between High Availability (HA) and Cluster mode?

-- TO UPDATE -- 

> A Redis Cluster consists of primary and secondary nodes. The cluster uses hash partitioning to split the keyspace into key slots. Each secondary node replicates a specific primary node and can be reassigned to replicate another primary or be elected to a primary node as needed. This is much better for scaling as the operation is spread across multiple nodes instead of having a single entry point.

Reviewer: same as precedent comment. primary/secondary to be clarified with the team

> A 2 nodes architecture for Redis™ with a primary node being duplicated in a standby node might increase the durability of the data but does not prevent the service from failing as automatic failover is not available for Redis databases. This means that the primary node will be a Single Point Of failure. Scaleway does not provide HA Database for Redis Instances but focuses instead on having a state-of-the-art Redis Cluster Architecture which is able to scale vertically and horizontally with built-in high availability. 

Reviewer: ```suggestion
A two nodes architecture for Redis™ with a primary node being duplicated in a standby node might increase the durability of the data but does not prevent the service from failing as automatic failover is not available for Redis databases. This means that the primary node will be a Single Point Of failure. Scaleway does not provide two nodes HA Database for Redis Instances but offers instead a state-of-the-art Redis Cluster Architecture which is able to scale vertically and horizontally with built-in High Availability. 
```

---
meta:
  title: Databases for Redis™
  description: Discover Scaleway Managed Databases and how to access it for Redis™.
content:
  h1: Databases for Redis™
  paragraph: Discover Scaleway Managed Databases and how to access it for Redis™.
image: database-redis.webp
---

## What is Scaleway Databases for Redis™

Databases for Redis™ is a low-latency caching solution. It allows you to easily set up a secure cache and lighten the load on your main database. Based on the in-memory data storage, Databases for Redis™ improves your application response time and helps you provide a better experience to your users. 

> The resource optimizes the speed of your requests as copies of the most frequently used data are stored in memory making them accessible in milliseconds. Learn more about Redis™ databases.

Author: ```suggestion
Using Databases for Redis™ as a cache optimizes the speed of your requests as copies of the most frequently used data are stored in memory, making them accessible in milliseconds. [Learn more about Redis™ databases](ADD_LINK_HERE).
```

> Using Databases for Redis™ as a cache optimizes the speed of your requests as copies of the most frequently used data are stored in memory, making them accessible in milliseconds. [Learn more about Redis™ databases](ADD_LINK_HERE).

Reviewer 2: ```suggestion
Using Databases for Redis™ as a cache optimizes the speed of your requests as copies of the most frequently used data are stored in memory, making them accessible in milliseconds. [Learn more about Databases for Redis™](ADD_LINK_HERE).
```

One of the most common ways to implement cache is storing frequently accessed data in Redis™ (therefore in memory), and serving your application's request from there, should data be available, instead of from your traditional database. If data is not available in memory, it can always be retrieved from the primary database. Below is a simplified schema explaining this process . 

## How can I access my Redis™ once it is provisioned?

You can connect to your Redis™  using the IP address.es and port.s available in the details page of your Redis™ Instance next to the title **Endpoints**.
- There are as many `IP:port` as nodes in your cluster (one if standalone).
- If you are in cluster mode (3 nodes or more), you have to use a cluster specific connector cable to connect to multiple endpoints.

ADD SCREENSHOT.

## What is the Private Networks feature for Redis™ Instances?

[Private Networks](https://www.scaleway.com/en/docs/network/vpc/quickstart/) allows you to configure your own Layer-2 network with IPv4 addresses on Databases for Redis™ Instances. 

> You can therefore enhance the security of your system architecture by isolating it from the Internet. It also increases the security of your databases, since Instances in your Private Network can directly be connected to your Redis™ Instance, without passing through the public Internet.

Author: ```suggestion
It also increases the security of your databases, since Instances in your Private Network can directly be connected to your Redis™ Instance, without passing through the public Internet.
```

In standalone mode, you can attach Private Networks to your Redis™ Instances at the creation level or add a Private Networks endpoint to your existing Redis™ Instance from the console.

<Message type="note">

> Bear in mind that in cluster mode, Private Network is only available at the creation of the cluster Instance and cannot be modified afterwards. If you want to switch back to a public network or update your IP adresses, you have to create a new Redis™ Instance. 

Reviewer 2: ```suggestion
Bear in mind that in cluster mode, Private Network is only available at the creation of the cluster Instance and cannot be modified afterwards. If you want to switch back to a public network or update your IP addresses, you have to create a new Redis™ Database Instance. 
```

</Message>

## How many Private Networks can I attach?
 
Currently, you can connect one Private Network to each Redis™ Instance in the Scaleway Console. For Standalone Instances you can attach multiple Private Networks via the CLI or the API only.

## Can I use any IP Range for my Private Network?
 

> For now, only the [RFC1918](https://datatracker.ietf.org/doc/html/rfc1918) IP subnets are accepted as a Redis™ Instance virtual IPv4.

Reviewer 2: ```suggestion
For now, only the [RFC1918](https://datatracker.ietf.org/doc/html/rfc1918) IP subnets are accepted as a Redis™ Database Instance virtual IPv4.
```

You can scale your cluster horizontally up to six nodes. Below is an example of configuration for a three nodes cluster:
* Instance A contains hash slots from 0 to 5,500
* Instance B contains hash slots from 5,501 to 11,000
* Instance C contains hash slots from 11,001 to 16,383

Each of the three Instances acts as a primary node and replicates one of the others as a secondary node.

## What's the difference between High Availability (HA) and Cluster mode?

-- TO UPDATE -- 

> A Redis™ Cluster consists of primary and secondary nodes. The cluster uses hash partitioning to split the keyspace into key slots. Each secondary node replicates a specific primary node and can be reassigned to replicate another primary or be elected to a primary node as needed. This is much better for scaling as the operation is spread across multiple nodes instead of having a single entry point.

Reviewer 2: ```suggestion
A Redis™ cluster consists of primary and secondary nodes. The cluster uses hash partitioning to split the keyspace into key slots. Each secondary node replicates a specific primary node and can be reassigned to replicate another primary or be elected to a primary node as needed. This is much better for scaling as the operation is spread across multiple nodes instead of having a single entry point.
```

> A 2 nodes architecture for Redis™ with a primary node being duplicated in a standby node might increase the durability of the data but does not prevent the service from failing as automatic failover is not available for Redis™ databases. This means that the primary node will be a Single Point Of failure. Scaleway does not provide HA Databases for Redis™ Instances but focuses instead on having a state-of-the-art Redis™ Cluster Architecture which is able to scale vertically and horizontally with built-in high availability. 

Reviewer 2: ```suggestion
A 2 nodes architecture for Redis™ with a primary node being duplicated in a standby node might increase the durability of the data but does not prevent the service from failing as automatic failover is not available for Redis™ databases. This means that the primary node will be a Single Point Of Failure. Scaleway does not provide HA Databases for Redis™ Instances but focuses instead on having a state-of-the-art Redis™ cluster architecture that is able to scale vertically and horizontally with built-in high availability. 
```
